framework/src/media/audio/audio_manager.c

```diff
@@ -867,102 +867,102 @@ audio_manager_result_t pause_audio_stream_out(void)
 	return AUDIO_MANAGER_SUCCESS;
 }
 
-audio_manager_result_t reset_audio_stream_in(void)
+audio_manager_result_t stop_audio_stream_in(void)
 {
+	int ret;
+
 	if (g_actual_audio_in_card_id < 0) {
 		meddbg("Found no active input audio card\n");
 		return AUDIO_MANAGER_NO_AVAIL_CARD;
 	}
 
 	pthread_mutex_lock(&(g_audio_in_cards[g_actual_audio_in_card_id].card_mutex));
 
-	pcm_close(g_audio_in_cards[g_actual_audio_in_card_id].pcm);
-	g_audio_in_cards[g_actual_audio_in_card_id].pcm = NULL;
-
-	if (g_audio_in_cards[g_actual_audio_in_card_id].resample.necessary) {
-		if (g_audio_in_cards[g_actual_audio_in_card_id].resample.buffer) {
-			free(g_audio_in_cards[g_actual_audio_in_card_id].resample.buffer);
-		}
-		src_destroy(g_audio_in_cards[g_actual_audio_in_card_id].resample.handle);
+	ret = pcm_drop(g_audio_in_cards[g_actual_audio_in_card_id].pcm);
+	if (ret < 0) {
+		meddbg("pcm_drop Failed, ret = %d\n", ret);
+		pthread_mutex_unlock(&(g_audio_in_cards[g_actual_audio_in_card_id].card_mutex));
+		return AUDIO_MANAGER_DEVICE_FAIL;
```

Comment:
How about moving pthread_mutex_unlock before meddbg isn't it?

Reply:
The unlock() is moved.